recordings-downloader/csharp/Recordings/Program.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using PureCloudPlatform.Client.V2.Api;
+using PureCloudPlatform.Client.V2.Client;
+using PureCloudPlatform.Client.V2.Extensions;
+using PureCloudPlatform.Client.V2.Extensions.Notifications;
+using PureCloudPlatform.Client.V2.Model;
+using Newtonsoft.Json;
+
+namespace Recordings
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // OAuth input
+            Console.Write("Enter Client ID: ");
+            string clientId = Console.ReadLine();
+            Console.Write("Enter Client Secret: ");
+            string clientSecret = Console.ReadLine();
+
+            // Get the Date Interval
+            Console.Write("Enter Date Interval (YYYY-MM-DDThh:mm:ss/YYYY-MM-DDThh:mm:ss): ");
+            string interval = Console.ReadLine();
+
+            Console.WriteLine("Working...");
+
+            // Configure SDK Settings
+            string accessToken = GetToken(clientId, clientSecret);
+            PureCloudPlatform.Client.V2.Client.Configuration.Default.AccessToken = accessToken;
+
+            // Instantiate APIs
+            ConversationsApi conversationsApi = new ConversationsApi();
+            RecordingApi recordingApi = new RecordingApi();
+
+            // Get Conversation IDs and corresponding Media Type within a date interval
+            Dictionary<string, AnalyticsSession.MediaTypeEnum> conversationsMapIDType = GetConversations(interval, conversationsApi);
+
+            // Use the Conversation IDs and request a batch download for the conversations
+            string jobId = RequestBatchDownload(conversationsMapIDType.Keys.ToList(), recordingApi);
+
+            // Get all conversations with recordings
+            List<BatchDownloadJobResult> jobResults = GetJobResults(jobId, recordingApi);
+
+            // Download all the recordings and separate into folders named after the Conversation ID
+            Console.WriteLine("Currently Downloading...");
+            string path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            path = path.Substring(0, path.LastIndexOf("bin"));
+
+            foreach (var job in jobResults)
+            {
+                // Create the directories for each unique Conversation IDs
+                System.IO.Directory.CreateDirectory(path + "\\recordings\\" + job.ConversationId);
+
+                // Determine the file extension to assign
+                string extension = GetExtensionFromMediaType(conversationsMapIDType[job.ConversationId]);
```

Comment:
Can you use the contentType from the batch job to determine the extension? I think that would give more accuracy for determining the file type because audio recordings are not always wav files. I think that would be `job.contentType` here.

Reply:
Currently, I am only familiar with audio/ogg for voice and application/zip for SMS. Do you happen to know other possible content types? Or any reference where I can view these content types.

Reply:
I'm not aware of this being documented anywhere. I've posted a question in the _PureCloud QM Dev_ chat group and tagged you.

Reply:
I did some work around with these. Please see the updated commit. For the mean time I will only integrate the Javascript/NPM part. The update for other languages will be available once done.